src/parser/jobs/sch/modules/DoTs.js

```diff
@@ -28,12 +29,140 @@ export default class DoTs extends CoreDoTs {
 		...DoTs.dependencies,
 		'checklist',
 		'suggestions',
+		'downtime',
 	]
 
 	static statusesToTrack = [
 		STATUSES.BIOLYSIS.id,
 	]
 
+	_lastBioCast = undefined
+	_application = {}
+
+	constructor(...args) {
+		super(...args)
+
+		const castFilter = {
+			by: 'player',
+			abilityId: ACTIONS.BIOLYSIS.id,
+		}
+		this.addEventHook('cast', castFilter, this._onTrackedDotCast)
+		const statusFilter = {
+			by: 'player',
+			abilityId: [STATUSES.BIOLYSIS.id],
+		}
+		this.addEventHook(['applydebuff', 'refreshdebuff'], statusFilter, this._onTrackedDotApply)
+	}
+
+	_pushApplication(targetKey, statusId, event) {
+		const target = this._application[targetKey] = this._application[targetKey] || {[STATUSES.BIOLYSIS.id]: []}
+		const source = this._lastBioCast
+		target[statusId].push({event, source})
+	}
+
+	_onTrackedDotCast(event) {
+		this._lastBioCast = event.ability.guid
```

Comment:
This isn't doing anything meaningful, since Biolysis's ID is the same all the time.  SMN tracks which skill applied the DOT since they have two different ways to apply it, but that doesn't really apply to SCH, so this should be removed, along with the cast hook and anything else using _lastBioCast.